File: m4d/ClientApp/src/models/ServiceUser.ts
import { jsonArrayMember, jsonMember, jsonObject } from "typedjson";
import { SimplePlaylist } from "./SimplePlaylist";

@jsonObject
export class ServiceUser {
  @jsonMember(String) public id!: string;
  @jsonMember(String) public name!: string;
  @jsonArrayMember(SimplePlaylist) public playlists?: SimplePlaylist[];
}

File: m4dModels/ServiceUser.cs
namespace m4dModels;

public class SimplePlaylist
{
    public string Id { get; set; }
    public string Name { get; set; }

    public string Description { get; set; }
    public int TrackCount { get; set; }
    public string Owner { get; set; }
    public string Music4danceId { get; set; }
}

public class ServiceUser
{
    public string Id { get; set; }
    public string Name { get; set; }
    public List<SimplePlaylist> Playlists { get; set; }
}
